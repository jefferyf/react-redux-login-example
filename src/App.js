import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginAction } from './actions/loginAction';
import logo from './logo.svg';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = { username: '', password: '' };
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    let { username, password } = this.state;
    this.props.loginAction(username, password);
    this.setState({ username: '', password: '' });
  }

  render() {
    let { username, password } = this.state;
    let { loginReducer: { isLoginPending, isLoginSuccess, loginError } } = this.props;
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          
          <form className="form" name="loginForm" onSubmit={this.onSubmit}>
            <div className="form-group">
              <label>Username</label>
              <input name="username" onChange={e => this.setState({username: e.target.value})} value={username}></input>
            </div>

            <div className="form-group">
              <label>Password</label>
              <input type="passsword" name="password" onChange={e => this.setState({password: e.target.value})} value={password}></input>
            </div>

            <input type="submit" value="Login"></input>
          </form>
          
          <div>
            <pre>{JSON.stringify(this.props)}</pre>
          </div>

          { isLoginPending && <div>Please wait...</div> }
          { isLoginSuccess && <div>Success.</div> }
          { loginError && <div>{loginError.message}</div> }
        </header>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  ...state
})

const mapDispatchToProps = dispatch => ({
  loginAction: (username, password) => dispatch(loginAction(username, password))
})

export default connect(mapStateToProps, mapDispatchToProps) (App);
