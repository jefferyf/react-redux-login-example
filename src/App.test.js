import React from 'react';
import { Provider } from 'react-redux';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import App from './App';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {};
  const mockStore = configureStore();
  const initialState = {
    // someState: [],
    // someReducer: {},
    loginReducer: {
      isLoginPending: false,
      isLoginSuccess: false,
      loginError: null
    }
  };

  const store = mockStore(initialState);

  const wrapper = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );

  return {
    props,
    wrapper    
  }
}

describe('snapshot tests', () => {
  it('renders correctly', () => {
    const { wrapper } = setup();

    const tree = renderer
      .create(wrapper)
      .toJSON();

    expect(tree).toMatchSnapshot();
  })
});

it('renders the login form', () => {
  const { wrapper } = setup();

  // console.log(enzymeWrapper.debug());
  expect(wrapper.find('.form').length).toBe(1);
});
