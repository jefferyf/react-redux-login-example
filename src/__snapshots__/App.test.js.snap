// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <form
      className="form"
      name="loginForm"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label>
          Username
        </label>
        <input
          name="username"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        <label>
          Password
        </label>
        <input
          name="password"
          onChange={[Function]}
          type="passsword"
          value=""
        />
      </div>
      <input
        type="submit"
        value="Login"
      />
    </form>
    <div>
      <pre>
        {"loginReducer":{"isLoginPending":false,"isLoginSuccess":false,"loginError":null}}
      </pre>
    </div>
  </header>
</div>
`;
